### PR TITLE
Implement `expand_to_increment` for `FixedDecimal`

### PR DESCRIPTION
**Pull request by jedel1043**
Next missing method from https://github.com/unicode-org/icu4x/issues/3929#issuecomment-1787872006. I think I covered all the edge cases with the tests, but let me know if you find anything else.

### CODE REVIEW

File: utils/fixed_decimal/src/decimal.rs [hunk 1]
```diff
@@ -1300,32 +1300,220 @@ impl FixedDecimal {
...
+
+                    // Must round the last digit to the next increment.
+                    let Some((_, digit)) = iter.next() else {
+                        debug_assert!(false, "`self.digits` should have at least a digit");
+                        return;
+                    };
+
+                    // Equivalent to (n + 2 / 2) * 2, which expands to the next
+                    // multiple of two
+                    *digit = (*digit + 2) & 0xFE;
```
Comment (sffc) [inline utils/fixed_decimal/src/decimal.rs:1415]: Note: When I first looked at this I questioned whether this should be `+1` instead of `+2` since only odd numbers are different between expand and trunc. I see now that `+2` is correct if there are additional trailing digits after the rounding position. However, I'm not completely sure that we handle cases where there are not trailing digits correctly. Can you check these cases?

| Input | Position | Increment | Expected Result |
|---|---|---|---|
| 0 | 0 | 2 | 0 |
| 0 | 0 | 5 | 0 |
| 0 | 0 | 25 | 0 |
| 0.1 | 0 | 2 | 2 |
| 0.1 | 0 | 5 | 5 |
| 0.1 | 0 | 25 | 25 |
| 1 | 0 | 2 | 2 |
| 1 | 0 | 5 | 5 |
| 1 | 0 | 25 | 25 |
| 2 | 0 | 2 | 2 |
| 2 | 0 | 5 | 5 |
| 2.1 | 0 | 2 | 4 |
| 2.1 | 0 | 5 | 5 |
| 4 | 0 | 2 | 4 |
| 4 | 0 | 5 | 5 |
| 4.1 | 0 | 2 | 6 |
| 4.1 | 0 | 5 | 5 |
| 5 | 0 | 2 | 6 |
| 5 | 0 | 5 | 5 |
| 5.1 | 0 | 2 | 6 |
| 5.1 | 0 | 5 | 10 |
| 6 | 0 | 2 | 6 |
| 6 | 0 | 5 | 10 |

File: utils/fixed_decimal/src/decimal.rs [hunk 2]
```diff
@@ -1300,32 +1300,220 @@ impl FixedDecimal {
...
+                    // last two digits removed from the iterator.
+
+                    *second_last_digit = 10;
+                    *last_digit = 10;
+
+                    let mut iter = self.digits.iter_mut().rev().enumerate();
+
+                    // Remove the last two items that were already handled.
+                    iter.next();
+                    iter.next();
```
Comment (sffc) [inline utils/fixed_decimal/src/decimal.rs:1508]: Suggestion (optional): to be more consistent with the other branches, consider using iterator methods to fetch `last_digit` and `second_last_digit` instead of `split_last_mut`. This is optional because what you currently have written is more consistent with how the `trunc` function is written.

File: utils/fixed_decimal/src/decimal.rs (hunk 1 above)
Comment (jedel1043) [reply utils/fixed_decimal/src/decimal.rs:1446]: Thanks for the test cases! I think this caught that bug, and I also used the checks in `trunc_to_increment` to avoid doing more work if the last digits are already a multiple of the increment.

File: utils/fixed_decimal/src/decimal.rs (hunk 2 above)
Comment (jedel1043) [reply utils/fixed_decimal/src/decimal.rs:1539]: I think I thought about doing it like that, but I opted for this because `iter` doesn't allow having two mutable references to its elements, so we'd have to recreate the iterator multiple times to first get the full number, then to modify the existing digits.

**Review by sffc — APPROVED** [review]
